Sources/Annotations/MKMapAnnotationView.swift

```diff
@@ -16,31 +16,51 @@ class MKMapAnnotationView<Content: View>: MKAnnotationView {
 
     private var controller: NativeHostingController<Content>?
 
-    // MARK: Methods
+    // MARK: Computed Properties
 
-    func setup(for mapAnnotation: ViewMapAnnotation<Content>) {
-        annotation = mapAnnotation.annotation
-        clusteringIdentifier = mapAnnotation.clusteringIdentifier
+    override var intrinsicContentSize: CGSize {
+        controller?.view.intrinsicContentSize
+            ?? .init(width: UIView.noIntrinsicMetric, height: UIView.noIntrinsicMetric)
+    }
 
-        let controller = NativeHostingController(rootView: mapAnnotation.content)
-        addSubview(controller.view)
-        bounds.size = controller.preferredContentSize
-        self.controller = controller
+    private var intrinsicContentFrame: CGRect {
+        let size = intrinsicContentSize
+        return CGRect(origin: .init(x: -size.width / 2, y: -size.height / 2), size: size)
     }
 
-    // MARK: Overrides
+    // MARK: Methods
 
-    #if canImport(UIKit)
+    func setup(for mapAnnotation: ViewMapAnnotation<Content>) {
+        #if canImport(UIKit)
+        backgroundColor = .clear
+        #endif
+
+        let controller = NativeHostingController(rootView: mapAnnotation.content, ignoreSafeArea: true)
 
-    override func layoutSubviews() {
-        super.layoutSubviews()
+        #if canImport(UIKit)
+        controller.view.backgroundColor = .clear
+        #endif
 
-        if let controller = controller {
-            bounds.size = controller.preferredContentSize
+        addSubview(controller.view)
+        
+        controller.view.translatesAutoresizingMaskIntoConstraints = false
+        let constraints = [
+            topAnchor.constraint(equalTo: controller.view.topAnchor),
+            leftAnchor.constraint(equalTo: controller.view.leftAnchor),
+            rightAnchor.constraint(equalTo: controller.view.rightAnchor),
+            bottomAnchor.constraint(equalTo: controller.view.bottomAnchor)
+        ]
+        NSLayoutConstraint.activate(constraints)
+        
+        self.controller = controller
+        self.invalidateIntrinsicContentSize()
+        
+        if let anchor = mapAnnotation.anchor {
+            centerOffset = .init(x: anchor.x * intrinsicContentFrame.width / 2, y: anchor.y * intrinsicContentFrame.height / 2)
```

Comment:
On iOS 16 you can just assign `anchorPoint` rather than calculating the `centerOffset`

Reply:
I don't know why, but `anchorPoint` does not really work in this case... I guess it may happen because of some size issues.

Reply:
I noticed another issue with `anchorPoint` - in the original implementation, the default anchorPoint is at (0.5, 0.5) whereas your implementation seems to use (0.0, 0.0) as default - could you please change that according to the original implementation to not cause confusion?

Reply:
@pauljohanneskraft 
Thank you for your response!
I changed my implementation according to the original SwiftUI version. Now, default anchorPoint is at (0.5, 0.5). Also, I renamed the `anchor` parameter to the `anchorPoint` as in original version.